### ISSUE TITLE
Cannot compile Pluto with latest version of Configurations.jl

### ISSUE THREAD

**Issue by zhangcx93** [Roger-luo/Configurations.jl#55]
Just used the ]up command, and Pluto failed to compile, due to updates of Configurations.jl

```
ERROR: The following 1 direct dependency failed to precompile:

Pluto [c3e4b0f8-55cb-11ea-2926-15256bba5781]

Failed to precompile Pluto [c3e4b0f8-55cb-11ea-2926-15256bba5781] to C:\Users\zhang\.julia\compiled\v1.6\Pluto\jl_6E.tmp.
ERROR: LoadError: LoadError: LoadError: LoadError: Evaluation into the closed module `Configurations` breaks incremental compilation because the side effects will not be permanent. This is likely due to some other module mutating `Configurations` with `eval` during precompilation - don't do this.
Stacktrace:
 [1] eval
   @ .\boot.jl:360 [inlined]
in expression starting at C:\Users\zhang\.julia\packages\Pluto\nkHsc\src\Configuration.jl:40
in expression starting at C:\Users\zhang\.julia\packages\Pluto\nkHsc\src\Configuration.jl:22
in expression starting at C:\Users\zhang\.julia\packages\Pluto\nkHsc\src\Configuration.jl:1
in expression starting at C:\Users\zhang\.julia\packages\Pluto\nkHsc\src\Pluto.jl:1
```

**Comment by zhangcx93** [comment]
By downgrade `]add Configurations@0.16.4` can fix it, so the latest 0.16.5 is causing the problem...

**Comment by zhubonan** [comment]
I got this too with a project depending on `Comonicon.jl` (happen in the CI so the latest version has been pulled). 

Related dependencies:
```
  [863f3e99] Comonicon v0.11.6
  [0acea522] ComoniconBuilder v0.1.4
  [a25bc8ac] ComoniconOptions v0.1.2
  [d6dfd36b] ComoniconTargetExpr v0.2.3
  [d0eb39ce] ComoniconTypes v0.2.0
  [dcf03e4a] ComoniconZSHCompletion v0.2.0
  [34da2185] Compat v3.39.0
  [5218b696] Configurations v0.16.5
  [187b0558] ConstructionBase v1.3.0
  [6add18c4] ContextVariablesX v0.1.2
```


Part of the stack trace:

```
ERROR: LoadError: LoadError: LoadError: LoadError: Evaluation into the closed module `Configurations` breaks incremental compilation because the side effects will not be permanent. This is likely due to some other module mutating `Configurations` with `eval` during precompilation - don't do this.
Stacktrace:
  [1] eval
    @ ./boot.jl:360 [inlined]
  [2] eval
    @ ~/.julia/packages/Configurations/RNhkY/src/Configurations.jl:1 [inlined]
  [3] is_reflect_type_expr(m::Module, ex::Any)
    @ Configurations ~/.julia/packages/Configurations/RNhkY/src/codegen.jl:144
  [4] has_duplicated_reflect_type
    @ ~/.julia/packages/Configurations/RNhkY/src/codegen.jl:135 [inlined]
  [5] option_m(mod::Module, ex::Expr, type_alias::Nothing)
    @ Configurations ~/.julia/packages/Configurations/RNhkY/src/codegen.jl:94
  [6] option_m(mod::Module, ex::Expr)
    @ Configurations ~/.julia/packages/Configurations/RNhkY/src/codegen.jl:92
  [7] var"@option"(__source__::LineNumberNode, __module__::Module, ex::Any)
    @ Configurations ~/.julia/packages/Configurations/RNhkY/src/codegen.jl:84
  [8] #macroexpand#50
    @ ./expr.jl:112 [inlined]
  [9] macroexpand
    @ ./expr.jl:111 [inlined]
 [10] docm(source::LineNumberNode, mod::Module, meta::Any, ex::Any, define::Bool) (repeats 2 times)
    @ Base.Docs ./docs/Docs.jl:537
 [11] var"@doc"(__source__::LineNumberNode, __module__::Module, x::Vararg{Any, N} where N)
    @ Core ./boot.jl:508
 [12] include(mod::Module, _path::String)
    @ Base ./Base.jl:386
 [13] include(x::String)
    @ ComoniconOptions ~/.julia/packages/ComoniconOptions/bW4Lz/src/ComoniconOptions.jl:1
 [14] top-level scope
    @ ~/.julia/packages/ComoniconOptions/bW4Lz/src/ComoniconOptions.jl:8
 [15] include
    @ ./Base.jl:386 [inlined]
 [16] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::String)
    @ Base ./loading.jl:1235
 [17] top-level scope
    @ none:1
 [18] eval
    @ ./boot.jl:360 [inlined]
 [19] eval(x::Expr)
    @ Base.MainInclude ./client.jl:446
 [20] top-level scope
    @ none:1
in expression starting at /home/runner/.julia/packages/ComoniconOptions/bW4Lz/src/types.jl:45
in expression starting at /home/runner/.julia/packages/ComoniconOptions/bW4Lz/src/types.jl:32
in expression starting at /home/runner/.julia/packages/ComoniconOptions/bW4Lz/src/types.jl:32
in expression starting at /home/runner/.julia/packages/ComoniconOptions/bW4Lz/src/ComoniconOptions.jl:1
ERROR: LoadError: Failed to precompile ComoniconOptions [a25bc8ac-9d24-42b2-8edd-3d267ec19fc5] to /home/runner/.julia/compiled/v1.6/ComoniconOptions/jl_K5e9kp.
```

**Comment by Roger-luo** [comment]
sorry about this, this is due to a recent release on checking `Maybe` type. should be fixed in #55
